litmus-portal/authentication/api/handlers/user_handlers.go

```diff
@@ -0,0 +1,176 @@
+			return
+		}
+		c.JSON(200, userResponse)
+	}
+}
+func UpdateUser(service user.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var userRequest entities.User
+		err := c.BindJSON(&userRequest)
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		userResponse, err := service.UpdateUser(&userRequest)
+		if err != nil {
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+		}
+		c.JSON(200, userResponse)
+	}
+}
+func FetchUsers(service user.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		users, err := service.GetUsers()
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+			return
+		}
+		c.JSON(200, users)
+	}
+}
+
+func LoginUser(service user.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var userRequest entities.User
+		err := c.BindJSON(&userRequest)
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		if userRequest.UserName == "" || userRequest.Password == "" {
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		userRequest.UserName = utils.SanitizeString(userRequest.UserName)
+		authenticatedUser, err := service.FindUser(&userRequest)
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidCredentials], presenter.CreateErrorResponse(utils.ErrInvalidCredentials))
+			return
+		}
+		token, err := authenticatedUser.GetSignedJWT()
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+			return
+		}
+		expiryTime := time.Duration(utils.JWTExpiryDuration) * 60
+		c.JSON(200, gin.H{
+			"access_token": token,
+			"expires_in":   expiryTime,
+			"type":         "Bearer",
+		})
+	}
+}
+
+func UpdatePassword(service user.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		username := c.MustGet("username").(string)
+		uid := c.MustGet("uid").(string)
+		var adminUser entities.User
+		adminUser.UserName = username
+		var userPasswordRequest entities.UserPassword
+		err := c.BindJSON(&userPasswordRequest)
+		userPasswordRequest.Username = username
+		adminUser.ID, _ = primitive.ObjectIDFromHex(uid)
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		if userPasswordRequest.OldPassword == "" || userPasswordRequest.NewPassword == "" {
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
```

Comment:
Should we put the password length/chars verification to the backend? " " password is a valid one according to this. Depending on frontend is not okay because we might have CLIs and external integrations so every check should be done on backend too.

Reply:
applies for every place that updates/sets the password

Reply:
is there any specific length of password length we need to abide by, or is it anything >= 2 characters?

Reply:
can you check the frontend, if it's not there we should add a standard one. Will have to discuss this if it's missing currently

Reply:
@gdsoumya , I checked the frontend repo. We apparently are not doing any extensive checks and are just checking if the length is >= 0.

https://github.com/litmuschaos/litmus/blob/302c7051a03d4f6a75d0425eb0601150479d3ffa/litmus-portal/frontend/src/views/Settings/AccountsTab/AccountSettings/index.tsx#L62-L75

Reply:
@DarthBenro008 Good to add standard checks for validation of username. For passwords let it be this way, we don't want to impose any hard checks on passwords as of now. cc: @gdsoumya

Reply:
Not sure if not imposing strict rules is a good idea need to discuss this

Reply:
@gdsoumya @SarthakJain26 So should we let this be as it is for the time being and create another issue related to this? As i think the conclusion of the discussion will require changes in both the places (the authentication back-end server as well as the frontend)

Reply:
We decided on adding a STRICT_PASSWORD_POLICY env var to the auth server, of this is set to true by the user the password policy will be used in the code else it will skip.

Reply:
By default it will be false. you will need to add the env var to the auth server manifest too

Reply:
@gdsoumya have we decided on the password policy? if so, could you please share it?

Reply:
I think this is good for now :
1. at least 8 char long
2. at least 1 special character, 1 uppercase ,1 lowercase letter and 1 digit

Abstract the password check into a diff function so that it can be updated independently later.